src/service/popup/popup.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Inject, Injectable, Injector, PLATFORM_ID } from '@angular/core';
+import { createCustomElement } from '@angular/elements';
+import { PopupComponent } from './popup.component';
+import { PopupType } from './popup.model';
+import { isPlatformBrowser } from '@angular/common';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PopupService {
+
+  constructor(private injector: Injector, @Inject(PLATFORM_ID) private platformId: Object) {
+    if (isPlatformBrowser(this.platformId)) {
+        const PopupElement = createCustomElement(PopupComponent, { injector });
+        if (!customElements.get('app-popup')) {
+          customElements.define('app-popup', PopupElement);
+        }   
+    }
+  }
+
+  showPopup(title: string, message: string, type: PopupType) {
```

Comment:
Trzy argumenty to jeszcze w granicach rozsądku, ale jeśli myślisz, że będziesz rozszerzał dalej ten komponent (np. będziesz chciał dodać opcje zanikania popupu po czasie) to sugeruję zamiast tego podawać tutaj obiekt konfiguracyjny a nie długą listę argumentów